cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -288,18 +285,16 @@ const char* castVARCHAR_utf8_int64(int64 context, const char* data, int32 data_l
     return data;
   }
 
-  int32_t char_count = utf8_length(context, data, data_len);
```

Comment:
utf8_length did validation on the input bytes. Since we are removing this, it won't do complete validation for utf8 characters